editor/js/Sidebar.Material.js

```diff
@@ -680,189 +688,224 @@ Sidebar.Material = function ( editor ) {
+		}
+
+		if ( material.envMap !== undefined ) {
+
+			materialEnvMapEnabled.setValue( material.envMap !== null );
+
+			if ( material.envMap !== null || resetTextureSelectors ) {
 
-				materialEnvMapEnabled.setValue( material.envMap !== null );
 				materialEnvMap.setValue( material.envMap );
-				materialReflectivity.setValue( material.reflectivity );
 
 			}
 
-			if ( material.lightMap !== undefined ) {
+			materialReflectivity.setValue( material.reflectivity );
+
+		}
+
+		if ( material.lightMap !== undefined ) {
+
+			materialLightMapEnabled.setValue( material.lightMap !== null );
+
+			if ( material.lightMap !== null || resetTextureSelectors ) {
 
-				materialLightMapEnabled.setValue( material.lightMap !== null );
 				materialLightMap.setValue( material.lightMap );
 
 			}
 
-			if ( material.aoMap !== undefined ) {
+		}
+
+		if ( material.aoMap !== undefined ) {
+
+			materialAOMapEnabled.setValue( material.aoMap !== null );
+
+			if ( material.aoMap !== null || resetTextureSelectors ) {
 
-				materialAOMapEnabled.setValue( material.aoMap !== null );
 				materialAOMap.setValue( material.aoMap );
-				materialAOScale.setValue( material.aoMapIntensity );
 
 			}
 
-			if ( material.side !== undefined ) {
+			materialAOScale.setValue( material.aoMapIntensity );
 
-				materialSide.setValue( material.side );
+		}
 
-			}
+		if ( material.side !== undefined ) {
 
-			if ( material.shading !== undefined ) {
+			materialSide.setValue( material.side );
 
-				materialShading.setValue( material.shading );
+		}
 
-			}
+		if ( material.shading !== undefined ) {
 
-			if ( material.blending !== undefined ) {
+			materialShading.setValue( material.shading );
 
-				materialBlending.setValue( material.blending );
+		}
 
-			}
+		if ( material.blending !== undefined ) {
 
-			if ( material.opacity !== undefined ) {
+			materialBlending.setValue( material.blending );
 
-				materialOpacity.setValue( material.opacity );
+		}
 
-			}
+		if ( material.opacity !== undefined ) {
 
-			if ( material.transparent !== undefined ) {
+			materialOpacity.setValue( material.opacity );
+
+		}
 
-				materialTransparent.setValue( material.transparent );
+		if ( material.transparent !== undefined ) {
 
-			}
+			materialTransparent.setValue( material.transparent );
 
-			if ( material.wireframe !== undefined ) {
+		}
 
-				materialWireframe.setValue( material.wireframe );
+		if ( material.wireframe !== undefined ) {
 
-			}
+			materialWireframe.setValue( material.wireframe );
 
-			if ( material.wireframeLinewidth !== undefined ) {
+		}
 
-				materialWireframeLinewidth.setValue( material.wireframeLinewidth );
+		if ( material.wireframeLinewidth !== undefined ) {
 
-			}
+			materialWireframeLinewidth.setValue( material.wireframeLinewidth );
+
+		}
 
-			updateRows();
+		setRowVisibility();
+
+	}
+
+	// events
+
+	signals.objectSelected.add( function ( object ) {
+
+		if ( object && object.material ) {
+
+			var objectChanged = object !== currentObject;
```

Comment:
Sure, please do. Here is why I did what I did:

Originally, I factored out <code>refreshUi</code> out of the select handler, in order to read back the new state into the UI when the material changes:
- <code>update</code> copies the state of the UI to the material instance, and
- <code>refreshUi</code> copies the state of the Material instance to the UI, where
- <code>refreshRows</code> takes care of structural changes.

So far, so good. It also allows us to react <code>materialChanged</code> signals originating from anywhere in the program by moving the direct call to <code>refreshUi</code> from the body of <code>update</code> to a signal handler. My previous message contains an error: It's not done that way yet (but at least, I obviously had that case in mind) and not needed for the script editor at all (<code>Script.js</code> sets the material state directly and there's nothing to display in the sidebar).
Note that I did not bother to conditionally read back the state after setting it for that reason. Also, it *should* be invariant, so let's see it break as obviously as possible (I usually prefer bugs and subtleties to show up, rather than letting them hide between the lines).

As things aren't always perfectly clean, it did break - because of the subtle ON/OFF state of the texture images, which has no counterpart in the data model.
As the GUI is now, it actually belongs into the material where it would be most impractically placed from the developer's point of view, from which it would best fit into texture, next to the sampler properties. However, the editor does not yet provide any means for editing those parts of the data model and they're still in flux...

So, my last commit fixes an issue that stems from a good enough, pragmatic design decision in a (hopefully) good enough, pragmatic way. Or more bluntly put: It's a bit of a hack :grin: At least an indication of the underying subtely is added to the code - maybe should have added a comment.